optaplanner-examples/src/test/java/org/optaplanner/examples/common/app/SolveAllTurtleTest.java

```diff
@@ -32,24 +36,36 @@
 /**
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
-@RunWith(Parameterized.class)
 public abstract class SolveAllTurtleTest<Solution_> extends AbstractTurtleTest {
 
+    interface ProblemFactory<Solution_> extends Function<File, Solution_> {
+
+        default Solution_ loadProblem(File f) {
+            return apply(f);
+        }
+    }
+
     private static final String MOVE_THREAD_COUNT_OVERRIDE = System.getProperty(TestSystemProperties.MOVE_THREAD_COUNT);
 
-    private final String solverConfigResource;
+    protected abstract List<File> getSolutionFiles(CommonApp<Solution_> commonApp);
 
-    public SolveAllTurtleTest(String solverConfigResource) {
-        this.solverConfigResource = solverConfigResource;
-    }
+    protected abstract CommonApp<Solution_> createCommonApp();
 
-    protected abstract Solution_ readProblem();
+    protected abstract ProblemFactory<Solution_> createProblemFactory(CommonApp<Solution_> commonApp);
 
-    @Test
-    public void runFastAndFullAssert() {
+    @TestFactory
+    Stream<DynamicTest> runFastAndFullAssert() {
         checkRunTurtleTests();
```

Comment:
Since the check is made here in the factory method, we have 1 skipped test for each turtle test class instead of skipping individual dynamic tests.

I just realized we can improve this even further by simply returning an empty stream from the test factory if turtle tests are not enabled. I think this is a nice use of the [dynamic test](https://junit.org/junit5/docs/current/user-guide/#writing-tests-dynamic-tests) feature.

Reply:
Thank you for pointing out. Makes sense.
Make sure that the skipped tests count comparison take this into account and ignores that difference - that's fine.